### PR TITLE
[Fixes JIRA #GEOS-8650] Externalize Importer working directory

### PR DESCRIPTION
**Pull request by afabiani**

### CODE REVIEW

File: src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java [hunk 1]
```diff
@@ -109,6 +114,14 @@
 public class Importer implements DisposableBean, ApplicationListener {
 
     static Logger LOGGER = Logging.getLogger(Importer.class);
+    
+    public static final String PROPERTYFILENAME = "geoserver-importer.properties";
```
Comment (aaime) [inline src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java]: Other plugins using simple config files are not prefixing with "geoserver-", like "controlflow.properties".
Probably better to just call it "importer.properties"

File: src/extension/importer/core/src/test/java/org/geoserver/importer/ImporterDataTest.java [hunk 2]
```diff
@@ -82,6 +83,32 @@ public SimpleFeature apply(ImportTask task, DataStore dataStore, SimpleFeature o
...
+        // Read the upload root folder and creates it if does not exists
+        importer.getUploadRoot();
+        
+        // Read the folder again...
+        dirFromProperties = Resources.directory(Resources.fromPath("props_uploads"));
+        // ... and ensure it is the same as defined on the .properties file
+        assertEquals(dirFromProperties, importer.getUploadRoot());
+        
+        // Let's now override the external folder through the Environment variable. This takes precedence on .properties
+        System.setProperty(Importer.UPLOAD_ROOT_KEY, "env_uploads");
```
Comment (aaime) [inline src/extension/importer/core/src/test/java/org/geoserver/importer/ImporterDataTest.java]: A test setting an env variable should be clearing it at the end (finally block here, or @AfterTest method)

File: src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java [hunk 3]
```diff
@@ -1645,7 +1678,36 @@ public File getImportRoot() {
...
+        }
+        
+        if (props != null && value == null) {
+            value = props.getProperty(UPLOAD_ROOT_KEY);
+        }
+        
+        try {
+            if(value != null) {
+                Resource uploadsRoot = Resources.fromPath(value);
+                return Resources.directory(uploadsRoot, !Resources.exists(uploadsRoot));
```
Comment (aaime) [inline src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java]: What if someone wants to use an absolute path to keep data outside of the data directory? (quite common)
Suggestion: use a fixed location relative to the data directory by default, and for configured locations, check first if they are absolute paths.

File: src/extension/importer/core/src/test/java/org/geoserver/importer/ImporterTestSupport.java [hunk 4]
```diff
@@ -74,6 +74,11 @@ protected void onSetUp(SystemTestData testData) throws Exception {
         namespaces.put("wms", "http://www.opengis.net/wms");
         
         XMLUnit.setXpathNamespaceContext(new SimpleNamespaceContext(namespaces));
+        
+        // add limits properties file
```
Comment (aaime) [inline src/extension/importer/core/src/test/java/org/geoserver/importer/ImporterTestSupport.java]: Limits?

Comment (aaime) [reply src/extension/importer/core/src/test/java/org/geoserver/importer/ImporterTestSupport.java]: The test should also check that the imported data actually ends up in that directory (either here or in a REST test)

File: src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java (hunk 1 above)
Comment (afabiani) [reply src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java]: Ok. Also might be good to have the importer config directory as you suggested on the other PR.

File: src/extension/importer/core/src/test/java/org/geoserver/importer/ImporterDataTest.java (hunk 2 above)
Comment (afabiani) [reply src/extension/importer/core/src/test/java/org/geoserver/importer/ImporterDataTest.java]: Right, I will fix it.

File: src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java (hunk 3 above)
Comment (afabiani) [reply src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java]: This check is done already by "Resources.fromPath" method.

On my tests it works either with relative and absolute paths.

File: src/extension/importer/core/src/test/java/org/geoserver/importer/ImporterTestSupport.java (hunk 4 above)
Comment (afabiani) [reply src/extension/importer/core/src/test/java/org/geoserver/importer/ImporterTestSupport.java]: The comment is wrong, will change it.

I'll add REST test also.

**Pull request merged**